pkg/api/apis/operators/v1alpha1/clusterserviceversion.go

```diff
@@ -98,7 +98,7 @@ func NewInstallModeSet(modes []InstallMode) (InstallModeSet, error) {
 // the given operatorNamespace and list of target namespaces.
 func (set InstallModeSet) Supports(operatorNamespace string, namespaces []string) error {
 	numNamespaces := len(namespaces)
-	if !set[InstallModeTypeAllNamespaces] && numNamespaces == 1 && namespaces[0] == v1.NamespaceAll {
+	if !set[InstallModeTypeAllNamespaces] && numNamespaces > 0 && namespaces[0] == v1.NamespaceAll {
```

Comment:
Can you explain this change? Why does it help to list additional namespaces for `NamespaceAll`?

Reply:
The exact scenario that is solved by doing this is after an operator group's status is written that is watching all namespaces, a new namespace creation isn't "noticed" without specifically listing all the namespaces seen.

Reply:
When an operator group is selecting all namespaces, the set of namespaces in its status should be of length = 1 and the first and only element should be `NamespaceAll` (""). There should never be a situation where length > 1 and the first element is "".